packages/color-modes/src/custom-properties.ts

```diff
@@ -70,19 +70,28 @@ export const objectToVars = (parent: string, obj: Record<string, any>) => {
 
 // create body styles for color modes
 export const createColorStyles = (theme: Theme = {}) => {
-  if (!theme.colors || theme.useBodyStyles === false) return {}
-  if (theme.useCustomProperties === false || !theme.colors.modes) {
+  const {
+    colors = {},
+    colors: { modes = {} } = {},
+    config: {
+      useBodyStyles,
+      useCustomProperties,
+      initialColorModeName,
+      printColorModeName,
+    } = {},
+  } = theme
```

Comment:
If anyone has suggestions for making this not be so ugly, I’m all-ears. I don’t have a lot of TypeScript experience & this solved all the errors I was running into, but it’s not the best.

Reply:
Why is this ugly? :P